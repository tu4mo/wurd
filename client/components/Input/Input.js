import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { media } from '../../utils/style'

const StyledInputError = styled.div`
  color: var(--color-red);
`

const Input = props => {
  const {
    className,
    color,
    error,
    id,
    maxLength,
    onChange,
    placeholder,
    type,
    value,
    ...rest
  } = props

  return (
    <div className={className}>
      <input
        {...rest}
        id={id}
        maxLength={maxLength}
        onChange={onChange}
        placeholder={placeholder}
        type={type}
        value={value}
      />
      {error && <StyledInputError>{error}</StyledInputError>}
    </div>
  )
}

Input.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  error: PropTypes.string,
  id: PropTypes.string.isRequired,
  maxLength: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string
}

const StyledInput = styled(Input)`
  input {
    background-color: #fff;
    border: 2px solid transparent;
    border-radius: var(--radius-sm);
    display: block;
    font-family: inherit;
    font-size: var(--font-size-lg);
    height: 40px;
    outline: none;
    padding: 0 var(--spacing-sm);
    width: 100%;

    ${media.sm`
      font-size: var(--font-size-md);
    `};

    &::-webkit-input-placeholder {
      color: var(--color-light-gray);
    }

    &:focus {
      border-color: var(--color-light-gray);
    }

    ${props =>
      props.color === 'gray' &&
      css`
        background-color: var(--color-background);
      `};

    ${props =>
      props.color === 'none' &&
      css`
        background-color: transparent;
        border: 0;
        border-radius: 0;
        height: auto;
        padding: 0;
      `};
  }
`

export default StyledInput
