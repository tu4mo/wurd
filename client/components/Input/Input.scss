.input {
  background-color: #fff;
  border: 2px solid transparent;
  border-radius: var(--radius-sm);
  display: block;
  font-family: inherit;
  font-size: var(--font-size-lg);
  height: 40px;
  outline: none;
  padding: 0 var(--spacing-sm);
  width: 100%;

  &::-webkit-input-placeholder {
    color: var(--color-light-gray);
  }

  &:focus {
    border-color: var(--color-light-gray);
  }

  &--gray {
    background-color: var(--color-background);
  }

  &--none {
    background-color: transparent;
    border: 0;
    border-radius: 0;
    height: auto;
    padding: 0;
  }
}
