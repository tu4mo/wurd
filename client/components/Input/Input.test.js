import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Input from './Input'

it('renders correctly', () => {
  const tree = renderer.create(<Input id="test-id" />).toJSON()

  expect(tree).toMatchSnapshot()
})
