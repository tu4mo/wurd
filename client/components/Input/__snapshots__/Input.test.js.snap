// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 input {
  background-color: #fff;
  border: 2px solid transparent;
  border-radius: var(--radius-sm);
  display: block;
  font-family: inherit;
  font-size: var(--font-size-lg);
  height: 40px;
  outline: none;
  padding: 0 var(--spacing-sm);
  width: 100%;
}

.c0 input::-webkit-input-placeholder {
  color: var(--color-light-gray);
}

.c0 input:focus {
  border-color: var(--color-light-gray);
}

<div
  className="c0"
>
  <input
    id="test-id"
    maxLength={undefined}
    onChange={undefined}
    placeholder={undefined}
    type={undefined}
    value={undefined}
  />
</div>
`;
